File: Tp_ProjetBank/Compte.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Tp_ProjetBank
{
    class Compte
    {
        private int numeroCompte;
        private double solde;

        public Compte() {}

        public Compte(int numero, double solde)
        {
            this.numeroCompte = numero;
            this.solde = solde;
        }

        public int GetNumero()
        {
            return numeroCompte;
        }
        public void SetNumero(int numero)
        {
            this.numeroCompte = numero;
        }
        public double GetSolde()
        {
            return solde;
        }
        
        //Deprecated : utiliser Ajouter() ou Retirer
        public void SetSolde(double solde)
        {
            this.solde = solde;
        }

        public void Ajouter(double unMontant)
        {
            solde += unMontant;
        }
        public virtual void Retirer(double unMontant)
        {
            solde -= unMontant;
        }
        public override string ToString()
        {
            return GetType().Name + " " + numeroCompte + " : solde = " + solde;
        }
    }
}

File: Enonces-Formation/Exo16-Evenements/en16.evenements/src/main/java/fr/banque/Compte.java
package fr.banque;

/**
 * Ceci est la classe Compte. <br>
 */
public class Compte extends AbstractEntity {
	private static final long serialVersionUID = 1L;
	private Double solde;
	private String libelle;

	/**
	 * Constructeur de l'objet. <br>
	 * Le numero par defaut sera -1.
	 */
	public Compte() {
		this(null, null);
	}

	/**
	 * Constructeur de l'objet. <br>
	 *
	 * @param unNumero
	 *            le numero du compte
	 * @param unSoldeInitial
	 *            le solde initial du compte
	 */
	public Compte(Integer unNumero, Double unSoldeInitial) {
		super(unNumero);
		this.setSolde(unSoldeInitial);
	}

	/**
	 * Donne acces au solde du compte. <br>
	 *
	 * @return le solde du compte
	 */
	public Double getSolde() {
		return this.solde;
	}

	/**
	 * Fixe le solde du compte. <br>
	 *
	 * @param unSolde
	 *            le nouveau solde du compte
	 */
	protected final void setSolde(Double unSolde) {
		this.solde = unSolde;
	}

	/**
	 * Ajoute un montant au compte. <br>
	 *
	 * @param unMontant
	 *            le montant ajoute au compte
	 */
	public void ajouter(Double unMontant) {
		this.setSolde(Double.valueOf(this.getSolde().doubleValue() + unMontant.doubleValue()));
	}

	/**
	 * Retire un montant du compte. <br>
	 *
	 * @param unMontant
	 *            le montant retire du compte
	 * @throws BanqueException
	 *             si un probleme survient
	 */
	public void retirer(Double unMontant) throws BanqueException {
		this.setSolde(Double.valueOf(this.getSolde().doubleValue() - unMontant.doubleValue()));
	}

	/**
	 * Formatage du compte sous forme de String utilisable directement par
	 * System.out.println(..);. <br>
	 *
	 * @return une representation chainee de l'objet
	 */
	@Override
	public String toString() {
		StringBuilder sb = new StringBuilder();
		sb.append(super.toString());
		sb.append(" Solde=");
		sb.append(this.getSolde());
		sb.append(" Libelle=");
		sb.append(this.getLibelle());
		return sb.toString();
	}

	/**
	 * Recupere le libelle du compte.
	 *
	 * @return le libelle du compte.
	 */
	public String getLibelle() {
		return this.libelle;
	}

	/**
	 * Modifie le libelle du compte.
	 *
	 * @param aLibelle
	 *            le libelle du compte.
	 */
	public void setLibelle(String aLibelle) {
		this.libelle = aLibelle;
	}
}
